examples/java-basic-mvn/src/main/java/pineconeexamples/MinimalUpsertAndQueryExample.java

```diff
@@ -1,62 +1,85 @@
 package pineconeexamples;
 
-import io.pinecone.PineconeClient;
-import io.pinecone.PineconeClientConfig;
-import io.pinecone.PineconeConnection;
-import io.pinecone.PineconeConnectionConfig;
-import io.pinecone.PineconeException;
-import io.pinecone.QueryRequest;
-import io.pinecone.QueryResponse;
-import io.pinecone.UpsertRequest;
-import io.pinecone.UpsertResponse;
-
-import java.util.Arrays;
+import com.google.common.primitives.Floats;
+import io.pinecone.*;
+import io.pinecone.proto.*;
+
 
 public class MinimalUpsertAndQueryExample {
+    public static class Args {
+        public String apiKey = System.getProperty("pinecone.apikey", "example-api-key");
+        String indexName = System.getProperty("pinecone.indexName", "example-index-name");
+        String environment = System.getProperty("pinecone.environment",
+                "example-environment");
+        String projectName = System.getProperty("pinecone.projectName", "example-project-name");
+        String namespace = "test-ns";
+        int topK = 1;
+    }
 
-    public static void main(String[] args) {
-        String apiKey = System.getProperty("pinecone.apikey", "example-api-key");
-        String serviceName = System.getProperty("pinecone.service.name", "example-service-name");
-        String target = System.getProperty("pinecone.service.url_authority",
-                "example-service-url-authority"); // see PineconeConnectionConfig#serviceAuthority for how to obtain
 
+    public static void main(String[] cliArgs) {
         System.out.println("Starting application...");
 
+        Args args = new Args();
+
         PineconeClientConfig configuration = new PineconeClientConfig()
-                .withApiKey(apiKey);
+                .withApiKey(args.apiKey)
+                .withEnvironment(args.environment)
```

Comment:
I wish we could put a comment here pointing to some docs about how to get env/project name. Best candidate would probably be https://www.pinecone.io/docs/quickstart/ > "About your index endpoint", but maybe that's too likely to be moved in future.